include/stdexec/execution.hpp

```diff
@@ -5767,29 +5767,40 @@ namespace stdexec {
   // [execution.senders.consumers.sync_wait]
   // [execution.senders.consumers.sync_wait_with_variant]
   namespace __sync_wait {
+    template <class _Context>
+    concept __driveable_context = requires(_Context& __ctx) {
+      { __ctx.get_scheduler() } noexcept -> scheduler;
+      { __ctx.run() };
+      { __ctx.finish() };
+    };
```

Comment:
My gut tells me that we don't want to be in the business of specifying the shape for execution contexts. Maybe what we want is a drivable scheduler, which is a refinement of the scheduler concept with `run` and `finish`.

Reply:
I agree